src/protocol/decoder.js

```diff
@@ -156,7 +157,7 @@ module.exports = class Decoder {
     }
 
     const stringBuffer = this.buffer.slice(this.offset, this.offset + byteLength)
```

Comment:
Having done a bit of reading, I think we have a mixup of different types.

The protocol makes a distinction between nullable and not-nullable fields. You can have COMPACT_STRING or COMPACT_NULLABLE_STRING, where the difference is that in the nullable version a length of 0 means null whereas in the not-nullable version a length of 0 means an empty string (same for the other COMPACT types).

At the moment, `readUVarIntBytes` is being used for both cases (see the `errorMessage` and `responses.name` in the AlterPartitionReassignmentsResponse), which won't work properly. We'll need to ensure that we have `readUVarIntString` and `readUVarIntNullableString`, etc, so that we can differentiate between the two cases.

Reply:
Nevermind. I wrote some tests to see if I could help you out with this, and I now see that in practice this doesn't matter. On the wire, there's no difference between a compact string and a compact nullable string. The only difference is whether we should allow a 0 value or if we should throw an error when reading/writing such a value - but if we treat all these types as nullable, then the only issue is if we receive an invalid message and interpret that as null instead of throwing an error - not a big deal.